xarray/coding/times.py

```diff
@@ -752,11 +828,101 @@ def encode_cf_datetime(
         # we already covered for this in pandas-based flow
         num = cast_to_int_if_safe(num)
 
-    return (num, units, calendar)
+    if dtype is not None:
+        num = _cast_to_dtype_if_safe(num, dtype)
+
+    return num, units, calendar
+
+
+def _encode_cf_datetime_within_map_blocks(
+    dates: np.ndarray,
+    units: str,
+    calendar: str,
+    dtype: np.dtype,
+) -> np.ndarray:
+    num, *_ = _eagerly_encode_cf_datetime(
+        dates, units, calendar, dtype, allow_units_modification=False
+    )
+    return num
+
+
+def _lazily_encode_cf_datetime(
+    dates: T_ChunkedArray,
```

Comment:
Th `lazily_*` functions are the only place you should be using `T_ChunkedArray`. :+1:

(None of this really matters right now because both `T_DuckArray` and `T_ChunkedArray` are set to `Any`, but it will make it all work later.)